brian2/core/variables.py

```diff
@@ -165,6 +164,16 @@ def __init__(self, name, dimensions=DIMENSIONLESS, owner=None, dtype=None,
         #: Whether the variable is an array
         self.array = array
 
+        #: Mark the list of attributes that should not be considered for caching
+        #: of state update code, etc.
+        self._cache_irrelevant_attributes = ['owner']
```

Comment:
This could be a class attribute maybe? Not sure if it's worth doing but maybe slightly more logical.